lib/michelson-test-check

```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3
+
+# Usage: ./kmich run ... --search-final --output json | $0 <path-to-definition>
+#
+# Reads kmichelson json output for unit-tests and checks that all branches have succeeded.
+# Exits 127 if there is a stuck term. Otherwise, exits with the number of failing branches
+
+import json
+import os
+import pyk
+import sys
+
+def findSubtermsByLabel(label, term):
+    if not term['node'] in ['KSequence', 'KApply']: return []
+    if term['node'] == 'KApply':
+        if term['label'] == label: return [term]
+        ret = []
+        for subterm in term['args']:
+            ret += findSubtermsByLabel(label, subterm)
+        return ret
+    return []
+
+def parseKastTerm(jsonString):
+    return json.loads(jsonString)['term']
+
+def findFailingBranches(term):
+    success_count = 0
+    failures = []
+    stucks = []
+    for kCell in findSubtermsByLabel('<k>', term):
+        assert(kCell['arity'] == 1)
+        kSeq = kCell['args'][0]
+        assert(kSeq['node'] == 'KSequence')
+
+        if   kSeq['items'] == []:                          success_count += 1
+        elif kSeq['items'][0]['label'] != '#AssertFailed': stucks += [kSeq]
+        else:                                              failures += [kSeq]
+    return success_count, failures, stucks
```

Comment:
`pyk` has `splitConfigFrom`, which would give you a map which looks like:

```
(
{ 'K_CELL' : CONTENTS_OF_<k>_CELL
, 'BLAH_CELL': CONTENTS_OF_<blah>_CELL
}

,

<k> K_CELL </k> <blah> BLAH_CELL </blah>
)
```

That is, it separates a config into a pair of a substitution and a term with variables, such that the substitution applied to the term is the original config. It picks where to do the separation by looking for the first children which are non-cells.

Reply:
OK, I had missed this.